scheduler/cups-driverd.cxx

```diff
@@ -2914,3 +2927,74 @@ regex_string(const char *s)		/* I - String to compare */
 
   return (NULL);
 }
+
+void *
+cupsFileGetsCall(void *input)
+{
+  retval = cupsFileGets(((struct args*)input)->fp,((struct args*)input)->buf,((struct args*)input)->buflen);
+  return NULL;
+}
+
+/*
```

Comment:
OK, so cupsFileGetsT implements a timed out read using a thread? This is totally unnecessary as UNIX-based systems can use `select` or `poll` (easier) to wait for input on the file descriptor (retrieved using cupsFileNumber).